### ISSUE TITLE
get_url across mount points

### ISSUE THREAD

**Issue by wonkydonky** [bottlepy/bottle#1054]
```
app/
    app.py
    modules/
        mod1/
            mod1.py
            views/
                view1.html
        mod2/
            mod2.py
            views/
                view2.html
```

app.py
```
root = Bottle()
root.mount ("/mod1", mod1app)
root.mount ("/mod2", mod2app)
```

Question: can I use `get_url` from a template, to get a URL from any module? For example

view1.html
```
{{  get_url ("root.about")  }}
{{  get_url ("mod1.articles")  }}
{{  get_url ("mod2.shop")  }}
```

**Comment by wonkydonky** [comment]
@carc1n0gen yes, but I got this error `UndefinedError("'get_url' is undefined")`. After reading the Bottle source, I discovered that I can use `bottle.BaseTemplate.defaults['get_url'] = root.get_url`. This works, in the sense that I can get a url for `root`. However, this does not work across modules mounted with `.mount()`.

**Comment by carc1n0gen** [comment]
Sounds like you have your answer.  I'm still new to bottle but so I can't 100% confirm though

**Comment by wonkydonky** [comment]
Ideally, I think it should work like this
```
root = Bottle()

mod1 = Bottle(name='mod1')
mod2 = Bottle(name='mod2')

root.mount ("/mod1", mod1)
root.mount ("/mod2", mod2)

bottle.BaseTemplate.defaults['get_url'] = root.get_url
```
and used like this from a template
```
{{ get_url('about') }}        <-    get "about" route from base app
{{ get_url('mod1.user') }}    <-    get "user" route from mod1 app
```
If this functionality could be added to Bottle, I think it would be great.

**Comment by wonkydonky** [comment]
> Sounds like you have your answer.

No I don't have my answer, because I can't use `get_url` across modules. I can only use for 1 module.

**Comment by carc1n0gen** [comment]
Looks like `get_url` is a method on the Bottle class so it probably only looks at the routes for that bottle instance. [1]

👍 For making this feature request from me though.

[1] https://github.com/bottlepy/bottle/blob/master/bottle.py#L871

**Comment by carc1n0gen** [comment]
> Sounds like you have your answer.

>> No I don't have my answer, because I can't use get_url across modules. I can only use for 1 module.

But you question was:

> Question: can I use get_url from a template, to get a URL from any module? For example

Currently the answer to that question is `no`

**Comment by wonkydonky** [comment]
Oh, you're right. The answer is no. I'd like to suggest this feature though.
